experiments/dorop-20/metadata/parsers.py

```diff
@@ -0,0 +1,210 @@
+from datetime import datetime
+from pathlib import Path
+
+from metadata.element_adapter import ElementAdapter
+from metadata.exceptions import DataNotFoundError, MetadataFileNotFoundError
+from metadata.models import (
+    Actor, Asset, AssetFile, AssetFileUse, CommonMetadata, FileMetadataFile, FileMetadataFileType,
+    PreservationEvent, RecordStatus, RepositoryItem, StructMap, StructMapItem, StructMapType
+)
+
+from pydantic import ValidationError
+
+def apply_relative_path(path: Path, path_to_apply: Path) -> Path:
+    return (path / path_to_apply).resolve(strict=True).relative_to(Path.cwd())
+
+class PremisEventParser():
+
+    def __init__(self, elem: ElementAdapter):
+        self.elem: ElementAdapter = elem
+
+    def get_event(self) -> PreservationEvent:
+        event_identifier = self.elem.find(".//PREMIS:eventIdentifierValue").text
+        event_type = self.elem.find(".//PREMIS:eventType").text
+        event_datetime = self.elem.find(".//PREMIS:eventDateTime").text
+        event_detail = self.elem.find(".//PREMIS:eventDetail").text
+        actor_role = self.elem.find(".//PREMIS:linkingAgentIdentifierType").text
+        actor_address = self.elem.find(".//PREMIS:linkingAgentIdentifierValue").text
+        return PreservationEvent(
+            identifier=event_identifier,
+            type=event_type,
+            datetime=datetime.fromisoformat(event_datetime),
+            detail=event_detail,
+            actor=Actor(address=actor_address, role=actor_role)
+        )
+
+class CommonMetadataParser():
+
+    def __init__(self, metadata_path: Path):
+        self.metadata_path: Path = metadata_path
+        try:
+            self.text = self.metadata_path.read_text()
+        except FileNotFoundError as e:
+            raise MetadataFileNotFoundError from e
+
+    def get_metadata(self) -> CommonMetadata:
+        try:
+            return CommonMetadata.model_validate_json(self.text)
+        except ValidationError as e:
+            raise DataNotFoundError("Common metadata parsing failed") from e
+
```

Comment:
This is kinda a mini-experiment with Pydantic. Aside from my `@schema` difficulties, it seems pretty neat-o.